src/lib/context.ts

```diff
@@ -234,8 +246,18 @@ export class AppContext {
         return `${this.loggerPrefix} ${message} ${this.loggerPostfix}`.trim();
     }
 
-    private prepareExtra(extra: Dict) {
+    private prepareExtra(extra?: Dict) {
+        if (extra === undefined) {
+            return extra;
+        }
         const preparedExtra = this.utils.redactSensitiveKeys(extra);
         return Object.keys(preparedExtra).length ? preparedExtra : undefined;
     }
+
+    private mergeExtra(extraParent?: Dict, extraCurrent?: Dict) {
```

Comment:
From my point of vew 
```
mergeExtra(extraParent: Dict | undefined, extraCurrent: Dict | undefined)
```
is better because in your cases these are not optional parameters, you always send them